spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2545,6 +2545,18 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
         withInfo(join, "SortMergeJoin is not enabled")
         None
 
+      case join: CartesianProductExec if isCometOperatorEnabled(op.conf, "cross_join") =>
+        // TODO: Support CartesianProductExec with join condition after new DataFusion release
+        if (join.condition.isDefined) {
+          withInfo(op, "cross_join with a join condition is not supported")
+          return None
+        }
+        None
```

Comment:
Seems you don't create the protobuf object for CrossJoin?

Reply:
it is my mistake